97afef8e00a019d14405ee4864e93234200f2528 mdp: Initialize dma_update_worker for MIPI_CMD_PANEL
diff --git a/drivers/video/msm/mdp.c b/drivers/video/msm/mdp.c
index de91ade..3b3fe56 100644
--- a/drivers/video/msm/mdp.c
+++ b/drivers/video/msm/mdp.c
@@ -1970,6 +1970,8 @@ static int mdp_probe(struct platform_device *pdev)
 		break;
 
 	case MIPI_CMD_PANEL:
+		INIT_WORK(&mfd->dma_update_worker,
+			  mdp_lcd_update_workqueue_handler);
 #ifndef CONFIG_FB_MSM_MDP303
 		mfd->dma_fnc = mdp4_dsi_cmd_overlay;
 		mipi = &mfd->panel_info.mipi;
